hveto/core.py

```diff
@@ -70,8 +73,68 @@ def deadtime(self):
 
 # -- core methods ------------------------------------------------------------
 
-def find_max_significance(primary, auxiliary, channel,
-                          snrs, windows, livetime):
+def find_all_coincidences(triggers, channel, snrs, windows):
+    """Find the number of coincs between each auxiliary channel and the primary
+
+    Parameters
+    ----------
+    primary : `numpy.ndarray`
+        an array of times for the primary channel
+    auxiliary : `numpy.recarray`
+        an array of triggers for a set of auxiliary channels
+    snrs : `list` of `float`
+        the SNR thresholds to use
+    window : `list` of `float`
+        the time windows to use
+    """
+    windows = sorted(windows, reverse=True)
+    snrs = sorted(snrs)
+    coincs = dict((p, {}) for p in itertools.product(windows, snrs))
+
+    for i, x in enumerate(triggers):
```

Comment:
@duncanmmacleod: You might find you get some enhanced performance if you apply the SNR threshold outside the coinc loop. I say this because it's an open ended and monotonically decreasing threshold, and it will reduce the number of tests required per pass through the coinc loop. You can just use a mask for the trigger array.  This requires an outer loop over SNR thresholds, though.

If you're happy with how things are currently, I can take a stab at forking and making those changes.